crates/precompile/src/blob/mod.rs

```diff
@@ -0,0 +1,139 @@
+use crate::{Error, Precompile, PrecompileAddress, PrecompileResult, B160};
+use c_kzg::{bindings, Bytes32, Bytes48, CkzgError};
+use revm_primitives::hex_literal::hex;
+use sha2::{Digest, Sha256};
+
+pub mod kzg_settings;
+
+pub const POINT_EVALUATION: PrecompileAddress =
+    PrecompileAddress(ADDRESS, Precompile::Standard(run));
+
+const ADDRESS: B160 = crate::u64_to_b160(0x0A);
+const GAS_COST: u64 = 50_000;
+const VERSIONED_HASH_VERSION_KZG: u8 = 0x01;
+
+#[allow(dead_code)]
+const FIELD_ELEMENTS_PER_BLOB: u64 = 4096;
+
+/// The big-endian representation of the modulus.
+#[allow(dead_code)]
+const BLS_MODULUS: &[u8; 32] =
+    &hex!("73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001");
+
+/// `U256(FIELD_ELEMENTS_PER_BLOB).to_be_bytes() ++ BLS_MODULUS.to_bytes32()`
+const RETURN_VALUE: &[u8; 64] = &hex!(
+    "0000000000000000000000000000000000000000000000000000000000001000"
+    "73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001"
+);
+
+fn run(input: &[u8], gas_limit: u64) -> PrecompileResult {
+    if gas_limit < GAS_COST {
+        return Err(Error::OutOfGas);
+    }
+    if input.len() != 192 {
+        return Err(Error::BlobInvalidInputLength);
+    }
+
+    // Verify commitment matches versioned_hash
+    let commitment = &input[96..144];
+    let versioned_hash = &input[0..32];
+    if kzg_to_versioned_hash(commitment) != versioned_hash {
+        return Err(Error::BlobMismatchedVersion);
+    }
+
+    // Verify KZG proof
+    let commitment = as_bytes48(commitment);
+    let z = as_bytes32(&input[32..64]);
+    let y = as_bytes32(&input[64..96]);
+    let proof = as_bytes48(&input[144..192]);
+    if !verify_kzg_proof(commitment, z, y, proof) {
+        return Err(Error::BlobVerifyKzgProofFailed);
+    }
+
+    Ok((GAS_COST, RETURN_VALUE.to_vec()))
```

Comment:
Is the return value always the same?

Reply:
Yes
[Geth impl](https://github.com/ethereum/go-ethereum/blob/5b159498bb73eca1c2864c97811ce0375532762c/core/vm/contracts.go#L1128C46-L1128C46)